conda/install.py

```diff
@@ -385,7 +385,7 @@ def dist2pair(dist):
 def dist2quad(dist):
     channel, dist = dist2pair(dist)
     parts = dist.rsplit('-', 2) + ['', '']
-    return (parts[0], parts[1], parts[2], channel)
+    return (str(parts[0]), str(parts[1]), str(parts[2]), str(channel))
 
```

Comment:
Arguably I should leave this as is and change it only in the call-site that then presents the values to `env`.